metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -2031,4 +2044,25 @@ private DataPlatformInfo getDataPlatformInfo(Urn urn) {
     }
     return null;
   }
+
+  private void checkConditionalUpdate(EntityAspect oldAspect, ExtraIngestParams extraIngestParams) {
```

Comment:
For these error messages, it's better to assume the client knows what they are doing and has set the right preconditions, so we're validating the system state, not the contents of the request headers.

For these error messages to be useful, I'm thinking of it from the perspective of an operator that has just sent 5 conditional requests and one or two of them fail. So for each error they will need to know:

1. What URN+aspect was being updated?
2. What was the expected state?
3. What was the actual state?

Given this info, the client can decide how to proceed.

Reply:
I'll put an example below but we'd have to add parameters `urn` and `aspectName` to this method to cover the necessary details. We should also add the same detail wherever we `throw new PreconditionFailedException`.